### PR TITLE
🎊 Should provide num_heads/hidden_size when opt_level == 0 and only_onnxruntime is False

### PR DESCRIPTION
**Pull request by trajepl**
## Describe your changes
1. Should provide num_heads/hidden_size when opt_level == 0 and only_onnxruntime is False
as this issue https://github.com/microsoft/onnxruntime/issues/17254
2. disable search for inc example.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/passes/onnx/transformer_optimization.py [hunk 1]
```diff
@@ -97,6 +97,23 @@ def validate_search_point(
...
             logger.info("CPUExecutionProvider does not support GPU inference, please avoid to use use_gpu.")
             return False
+        if search_point.get("only_onnxruntime") and search_point.get("opt_level") <= 0:
+            logger.info("Please specify a positive value for opt_level when only_onnxruntime is True")
+            return False
+        if (
+            search_point.get("opt_level") == 0
+            and search_point.get("only_onnxruntime")
+            and search_point.get("num_heads") == 0
+            and search_point.get("hidden_size") == 0
```
Comment (guotuofeng) [inline olive/passes/onnx/transformer_optimization.py:107]: seems the ort optimize_by_fused allow 0 for bert model?

**Comment by guotuofeng** [comment]
what's the reason to disable search?

File: olive/passes/onnx/transformer_optimization.py (hunk 1 above)
Comment (trajepl) [reply olive/passes/onnx/transformer_optimization.py:107]: Yes, opt_level can be 0, but at the same time the num_heads and hidden_size cannot be 0 as this issue:
https://github.com/microsoft/onnxruntime/issues/17254

File: olive/passes/onnx/transformer_optimization.py [hunk 2]
```diff
@@ -97,6 +97,23 @@ def validate_search_point(
...
+        ):
+            from onnxruntime import __version__ as OrtVersion
+            from packaging import version
+
+            if version.parse(OrtVersion) <= version.parse("1.16.0"):
```
Comment (guotuofeng) [inline olive/passes/onnx/transformer_optimization.py:112]: will 1.16 fix this issue? need we add this check? 
BTW, do we need log as warnings?

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**